toolchain/check/interface_support.cpp

```diff
@@ -0,0 +1,86 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "toolchain/check/interface_support.h"
+
+#include "toolchain/check/inst.h"
+#include "toolchain/check/merge.h"
+#include "toolchain/check/name_lookup.h"
+#include "toolchain/sem_ir/entity_with_params_base.h"
+
+namespace Carbon::Check {
+
+auto GetExistingDeclOrDiagnoseMismatch(
```

Comment:
Nit: "OrDiagnoseMismatch" implies that it's only diagnosing mismatches, but you've made this do more. Maybe something drop "Mismatch", or rename to "TryGetExistingDecl"? I think there are other cases we just document as possibly diagnosing, and the caller is supposed to just be happy it doesn't need to deal with it.

Reply:
Done